daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -501,34 +501,37 @@ reset args = do
               (L.Verbosity False)
           let chunks = chunksOf 100 activeContracts
           forM_ chunks $ \chunk -> do
-            cmdId <- liftIO UUID.nextRandom
-            let cmds =
-                  L.Commands
-                    { coms =
-                        [ L.ExerciseCommand
-                          { tid = tid
-                          , cid = cid
-                          , choice = L.Choice "Archive"
-                          , arg = L.VRecord $ L.Record Nothing []
-                          }
-                        | (_offset, _mbWid, events) <- chunk
-                        , L.CreatedEvent {cid, tid} <- events
-                        ]
-                    , lid = ledgerId
-                    , wid = Nothing
-                    , aid = L.ApplicationId "ledger-reset"
-                    , cid = L.CommandId $ TL.fromStrict $ UUID.toText cmdId
-                    , actAs = parties
-                    , readAs = []
-                    , dedupPeriod = Nothing
-                    , minLeTimeAbs = Nothing
-                    , minLeTimeRel = Nothing
-                    , sid = Nothing
-                    }
-            errOrEmpty <- L.submit cmds
-            case errOrEmpty of
-              Left err -> liftIO $ putStrLn $ "Failed to archive active contracts: " <> err
-              Right () -> pure ()
+            let archiveCommands = [ L.ExerciseCommand
```

Comment:
If you wanted to make a smaller change, you could write `let cmds cmdId` instead of moving the definition for the purpose of being in scope of the binding of `cmdId`.